proxy/src/write.rs

```diff
@@ -425,6 +416,47 @@ impl<Q: QueryExecutor + 'static> Proxy<Q> {
             })
     }
 
+    async fn handle_auto_create_table(
+        &self,
+        request_id: RequestId,
+        database: &str,
+        req: &WriteRequest,
+    ) -> Result<()> {
+        if !self.auto_create_table {
+            return Ok(());
+        }
+
+        let schema_config = self
+            .schema_config_provider
+            .schema_config(database)
+            .box_err()
+            .with_context(|| ErrWithCause {
+                code: StatusCode::INTERNAL_SERVER_ERROR,
+                msg: format!("Fail to fetch schema config, schema_name:{database}"),
+            })?
+            .cloned()
+            .unwrap_or_default();
+        for write_table_req in &req.table_requests {
+            let table_info = self
+                .router
+                .fetch_table_info(database, &write_table_req.table)
+                .await?;
+            if table_info.is_some() {
+                continue;
+            }
+            self.create_table(
```

Comment:
This method will always get executed? Even for already existing tables?

Reply:
```
if table_info.is_some() {
                continue;
            }
```
Existing tables will skip creation.